script.js

```diff
@@ -1,4 +1,447 @@
+  }
+  // console log to track which L1 mode we are current in
+  console.log("L1GameMode after", L1GameMode);
+  console.log("arrayForNums", arrayForNums);
+  return myOutputValue;
+};
+
+//____________________________ Mode to select which game to play________________
+
+// Start game function where after user choose their game mode (Normal, Lowest Win, Variable Dice), we will set the L1 mode. L2 mode will prime the code to know which step to take.
+
+var startGame = function (input) {
+  var myOutputValue = `Please type in "Normal", "Lowest Win", "Auto", "Variable Dice", "Variable Player", "Knockout" `;
+  if (input == "Normal") {
+    L1GameMode = L1_NORMAL_MODE;
+    myOutputValue = `You have selected ${input} as your game of choice! <br><br>2 Players will take turns to roll the dice 🎲.<br><br>After the roll, each player may choose the order to form their number.<br><br>Player with the highest number wins.<br><br>Click Submit to start rolling!`;
+    L2GameMode = L2_ROLL_DICE;
+  }
+  if (input == "Lowest Win") {
+    L1GameMode = L1_LOWEST_MODE;
+    myOutputValue = `You have selected ${input} as your game of choice! <br><br>2 Players will take turns to roll the dice 🎲.<br><br>After the roll, each player may choose the order to form their number.<br><br>Player with the lowest number wins.<br><br>Click Submit to start rolling!`;
+    L2GameMode = L2_ROLL_DICE;
+  }
+  if (input == "Auto") {
+    L1GameMode = L1_AUTO_MODE;
+    myOutputValue = `You have selected ${input} as your game of choice! <br><br>2 Players will take turns to roll the dice 🎲.<br><br>After the roll, Computer will choose the best order for player.<br><br>Player with the highest number wins.<br><br>Click Submit to start rolling!`;
+    L2GameMode = L2_ROLL_DICE_AUTO;
+  }
+  if (input == "Variable Dice") {
+    L1GameMode = L1_VARIABLE_DICE_MODE;
+    myOutputValue = `You have selected ${input} as your game of choice! <br><br> Start by choosing the number of 🎲.`;
+    L2GameMode = L2_CHOOSE_NUM_DICE;
+  }
+  if (input == "Variable Players") {
+    L1GameMode = L1_VARIABLE_PLAYERS_MODE;
+    myOutputValue = `You have selected ${input} as your game of choice! <br><br> Start by choosing the number of players ⛹🏻‍♀️⛹🏻‍♂️.`;
+    L2GameMode = L2_CHOOSE_NUM_DICE;
+  }
+  return myOutputValue;
+};
+
+//_______________________________ Normal Beat That Mode _____________________
+
+// this is the basic Beat That Game
+
+// main function to determine which step to take
+
+var normal = function (input) {
+  var myOutputValue = "";
+  if (L2GameMode == L2_ROLL_DICE) {
+    myOutputValue = roll(input);
+  } else if (L2GameMode == L2_CHOOSE_ORDER) {
+    myOutputValue = pickOrder(input);
+  } else if (L2GameMode == L2_DECIDE_WINNER) {
+    myOutputValue = decideWinner(input);
+  }
+  return myOutputValue;
+};
+
+// this is to roll the number of dice to and give the output to user to pick. while loop was done to prepare for multiple dice game.
+
+var roll = function () {
```

Comment:
Highlighting line 162 & line 173 specifically.
In line 162, you are placing an argument within the roll function --> `roll(input)`
In line 173 though, you defined a function without a parameter --> `var roll = function **()** {`

Not sure if this was what you wanted? But I can see how you are calling the normal() function right at the top, which is great work